### PR TITLE
Add isValidIndex function

### LINKED ISSUE #1043: Add isValidIndex function
As in #1024 and #1042, should we add a general function to test if the input is a valid H3 index of any mode (cell, directed edge, vertex, etc.)?

The code would be something like:

```c
bool isValid = H3_EXPORT(isValidCell)(cell) ||
               H3_EXPORT(isValidDirectedEdge)(cell) ||
               H3_EXPORT(isValidVertex)(cell);
```

### PR DESCRIPTION
**Pull request by cnaples79**
## Summary
Implements a general validation function `isValidIndex()` that checks if an H3 index is valid for any mode (cell, directed edge, or vertex), as requested in #1043.

## Changes Made
- **API Addition**: Added `isValidIndex()` function declaration to `h3api.h.in`
- **Implementation**: Implemented function in `h3Index.c` that checks all three validation types:
  ```c
  int H3_EXPORT(isValidIndex)(H3Index h) {
      return H3_EXPORT(isValidCell)(h) ||
             H3_EXPORT(isValidDirectedEdge)(h) ||
             H3_EXPORT(isValidVertex)(h);
  }
  ```
- **Dependencies**: Added includes for `directedEdge.h` and `vertex.h` to `h3Index.c`
- **Tests**: Added comprehensive test coverage in `testH3Index.c` covering:
  - Valid cell indices
  - Valid directed edge indices
  - Valid vertex indices
  - Invalid indices
  - Corrupted indices

## API Usage
```c
H3Index index = ...; // Can be a cell, edge, or vertex
if (isValidIndex(index)) {
    // Index is valid for some mode
}
```

## Test Plan
- Added unit test `isValidIndex` in `testH3Index.c`
- Tests validate the function returns true for valid cells, directed edges, and vertices
- Tests validate the function returns false for invalid and corrupted indices
- All existing tests should continue to pass

Fixes #1043

### CODE REVIEW

File: src/apps/testapps/testH3Index.c [hunk 1]
```diff
@@ -212,4 +212,39 @@ SUITE(h3Index) {
...
+        LatLng coord = {0, 0};
+        H3Index cell;
+        t_assertSuccess(H3_EXPORT(latLngToCell)(&coord, 5, &cell));
+        t_assert(H3_EXPORT(isValidIndex)(cell),
+                 "isValidIndex returns true for valid cell");
+
+        // Test with valid directed edge
+        H3Index neighbor;
+        t_assertSuccess(H3_EXPORT(latLngToCell)(&coord, 5, &neighbor));
+        neighbor++;  // Get a neighboring cell
```
Comment (ajfriend) [inline src/apps/testapps/testH3Index.c]: This won't work. Simply incrementing will almost never give you a neighboring cell.

Comment (cnaples79) [reply src/apps/testapps/testH3Index.c]: @ajfriend I just pushed a commit to fix this. Thanks for the feedback. I used gridRingUnsafe to call the neighboring cells. Let me know what you think.

**Comment by ajfriend** [comment]
Thanks for this! It looks like you may need to run/test this locally to get the formatting that the CI is expecting.

**Comment by cnaples79** [comment]
@ajfriend happy to contribute. I fixed the formatting manually based on the logs then confirmed using clang-format. CI should hopefully pass now.

**Comment by cnaples79** [comment]
CI formatting failed on the header file, I re-formatted it with clang-format.

**Comment by ajfriend** [comment]
I'll land this now. Thanks @cnaples79!

I've added #1058, #1059, https://github.com/uber/h3/issues/1060, which anyone should feel free to take as a follow-up.
